### ISSUE TITLE
tsql: label following select without semicolon unparsable

### ISSUE THREAD

**Issue by ddbtrmatic** [sqlfluff/sqlfluff#7857]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using a label, i.e. `mylabel:` in tsql, if the label follows a select statement without a semicolon teminator, then it will be unparasable.

### Expected Behaviour

This is valid tsql syntax

```sql
BEGIN
    SELECT 1
    label:
END
```

### Observed Behaviour

```sql
BEGIN
    SELECT 1
    label:
END
```

```
INFO
###
#
# Fixed Tree:
#
###
INFO
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Sequence: [<AnyNumberOf: [<Ref: 'BatchSegment'>]>]>"
[L:  1, P:  1]      |        word:                                                 'BEGIN'
[L:  1, P:  6]      |        newline:                                              '\n'
[L:  2, P:  1]      |        whitespace:                                           '    '
[L:  2, P:  5]      |        word:                                                 'SELECT'
[L:  2, P: 11]      |        whitespace:                                           ' '
[L:  2, P: 12]      |        numeric_literal:                                      '1'
[L:  2, P: 13]      |        newline:                                              '\n'
[L:  3, P:  1]      |        whitespace:                                           '    '
[L:  3, P:  5]      |        word:                                                 'label'
[L:  3, P: 10]      |        colon:                                                ':'
[L:  3, P: 11]      |        newline:                                              '\n'
[L:  4, P:  1]      |        word:                                                 'END'
[L:  4, P:  4]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    [META] end_of_file:

== [label.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'BEGIN\n
                       | SELECT 1\n    label:\nEND'
```

Adding a semicolon results in the parsing succeeding.

```sql
BEGIN
    SELECT 1;
    label:
END
```

```
== [label.sql] PASS
```

### How to reproduce

`echo "BEGIN SELECT 1 label: END" > label.sql && sqlfluff lint --dialect tsql label.sql` fails

`echo "BEGIN SELECT 1; label: END" > label.sql && sqlfluff lint --dialect tsql label.sql` passes

### Dialect

tsql

### Version

sqlfluff, version 4.2.0

### Configuration

no configuration, default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by IdanGonen** [comment]
@ddbtrmatic @peterbud 

[PR Link](https://github.com/sqlfluff/sqlfluff/pull/7867)
